src/commands/mod.rs

```diff
@@ -2152,7 +2152,7 @@ from the source will be moved into the parent.
         args.interactive,
         matcher.as_ref(),
     )?;
-    if &new_parent_tree_id == parent.tree_id() {
+    if &new_parent_tree_id == parent.tree_id() && !commit.is_empty() {
```

Comment:
Wouldn't that deserve a message on the terminal saying that this `squash` operation was a no-op?

Reply:
And btw, in which case can this happen? Shouldn't the `if` block be completely removed?

Reply:
Interesting, I haven't considered the interactive case. @samueltardieu to my understanding, we need to consider the Cartesian product of the following states here: [empty commit, non-empty commit] x [description set, description not set] x [non-interactive, interactive: no changes selected, interactive: some changes selected, interactive: all changes selected]

1) empty + no desc + not int => abandon(no-op)
2) empty + no desc + int: nothing => Err
3) empty + no desc + int: some => impossible
4) empty + no desc + int: all => impossible
5) empty + desc + not int => abandon, ask
6) empty + desc + int: nothing => Err
7) empty + desc + int: some => impossible
8) empty + desc + int: all => impossible

9) not empty + no desc + not int => squash, abandon
10) not empty + no desc + int: nothing => Err
11) not empty + no desc + int: some => squash
12) not empty + no desc + int: all => squash, abandon
13) not empty + desc + not int => squash, ask
14) not empty + desc + int: nothing => Err
15) not empty + desc + int: some => squash, ask
16) not empty + desc + int: all => squash, ask, abandon

It is actually arguable what the behaviour in (2) and (6) should be, but I set it to Err for consistency with (10) and (14)

Also, it seems to me that what @martinvonz suggested makes total sense and it solves all the cases above. So I used that.

Reply:
As for the message - I can add it, but I wonder if it is helpful? The abandoned commit was empty, so why bother user?

Reply:
I agree, I don't see a reason to tell the user about it since nothing was lost.

Reply:
I hope I can mark this as resolved. Please feel free to reopen if there are any unanswered questions I missed.